### PR TITLE
 ngtcp2: Return code for QUIC Connect error 

### PR DESCRIPTION
**Pull request by emilengler**
Removes some todos.
Adds init code 96 and 97 for Quic (SSL) initialization errors

### CODE REVIEW

**Comment by jay** [comment]
I agree a change is needed but I would use a single error code like CURLE_QUIC_CONNECT_ERROR (in the spirit of CURLE_SSL_CONNECT_ERROR) in place of all the CURLE_FAILED_INIT in that file. The INIT error is intended for library initialization errors. This will remain open to gather feedback. It's the holidays and we're in a feature freeze so regardless this won't move forward right away.

**Comment by emilengler** [comment]
@jay I was also a bit skeptical as SSL is included into Quic...
Ok, I will change it the days.

Have a nice Christmas and a happy new year ;)

**Comment by emilengler** [comment]
@jay Done

**Comment by jay** [comment]
Thanks, Merry Christmas to you as well.

File: include/curl/curl.h [hunk 1]
```diff
@@ -609,6 +609,7 @@ typedef enum {
   CURLE_AUTH_ERROR,              /* 94 - an authentication function returned an
                                     error */
   CURLE_HTTP3,                   /* 95 - An HTTP/3 layer problem */
+  CURLE_QUIC_CONNECT_ERROR,      /* 96 - A QUIC Init problem */
```
Comment (jay) [inline include/curl/curl.h]: QUIC connect problem not init problem

**Comment by jay** [comment]
CIs are failing:

strerror.c: In function ‘curl_easy_strerror’:
strerror.c:60:3: error: enumeration value ‘CURLE_QUIC_CONNECT_ERROR’ not handled in switch [-Werror=switch]
   switch(error) {
   ^~~~~~

also you'll have to add it to
docs/libcurl/libcurl-errors.3
docs/libcurl/symbols-in-versions
packages/OS400/curl.inc.in

File: include/curl/curl.h (hunk 1 above)
Comment (emilengler) [reply include/curl/curl.h]: Oh forgot that comment, thanks

**Comment by emilengler** [comment]
> you can change the rest of the CURLE_FAILED_INIT in that file to CURLE_QUIC_CONNECT_ERROR

Done

**Comment by emilengler** [comment]
> packages/OS400/curl.inc.in
I'm not really sure about that file, not all error codes are included there. e.g `CURLE_HTTP3`

**Comment by jay** [comment]
> I'm not really sure about that file, not all error codes are included there. e.g `CURLE_HTTP3`

Yes they were missing. I've added them in master, rebased your branch on that and then added CURLE_QUIC_CONNECT_ERROR to os400 as well as a few other changes in description. I have just force-pushed them to the branch.

**Comment by emilengler** [comment]
@jay Thanks

**Comment by bagder** [comment]
I like the new error code. This PR needs a rebase.

**Comment by emilengler** [comment]
@bagder Rebase done

**Comment by jay** [comment]
Thanks

**Pull request closed without merging**